modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/components/columns/RolesColumn/components/RolesDropdown/RolesDropdown.jsx

```diff
@@ -18,84 +21,45 @@ const RolesDropdown = ({
 	onClick,
 	supportSeatsCount,
 }) => {
-	const [active, setActive] = useState(false);
-
+	const [radioOptions, setRadioOptions] = useState({});
 	const [selectedAccountRoleName, setSelectedAccountRoleName] = useState(
 		currentRoleBriefName
 	);
 
-	const items = useMemo(
-		() =>
-			accountRoles.map((accountRole) => ({
-				active: accountRole.name === selectedAccountRoleName,
-				disabled: hasAccountSupportSeatRole
-					? supportSeatsCount === 1
-					: isSupportSeatRole(accountRole.name) &&
-					  availableSupportSeatsCount === 0,
-				label: accountRole.name,
-				raysourceName: accountRole.raysourceName,
-				value: accountRole.id,
-			})),
-		[
-			accountRoles,
-			availableSupportSeatsCount,
-			hasAccountSupportSeatRole,
-			selectedAccountRoleName,
-			supportSeatsCount,
-		]
-	);
+	useEffect(() => {
+		const baseFormatAccount = accountRoles.map((accountRole) => ({
+			active: selectedAccountRoleName.includes(accountRole.name),
+			disabled: hasAccountSupportSeatRole
+				? supportSeatsCount === 1
+				: isSupportSeatRole(accountRole.name) &&
+				availableSupportSeatsCount === 0,
+			label: accountRole.name,
+			raysourceName: accountRole.raysourceName,
+			value: accountRole.id,
+		}))
+
+		setRadioOptions(baseFormatAccount.reduce((previousItem, item) => {
+			if (!partnerMemberRoles.includes(item.label)) {
+				previousItem[item.label] = item;
 
-	const handleOnClick = (accountRoleItem) => {
-		if (accountRoleItem.label !== selectedAccountRoleName) {
-			onClick(accountRoleItem);
-			setSelectedAccountRoleName(accountRoleItem.label);
-		}
-	};
+				return previousItem;
+			}
 
-	const getDropdownItems = () =>
-		items.map((item, index) => (
-			<DropDown.Item
-				className="pr-6"
-				disabled={item.disabled}
-				key={`${item.label}-${index}`}
-				onClick={() => handleOnClick(item)}
-				symbolRight={item.active && 'check'}
-			>
-				{i18n.translate(getKebabCase(item.label))
-					? i18n.translate(getKebabCase(item.label))
-					: item.label}
-			</DropDown.Item>
-		));
+			previousItem.partnerMemberRoles.roles.push(item);
+			previousItem.partnerMemberRoles.active = previousItem.partnerMemberRoles.active ? true : item.active
 
-	return (
-		<DropDown
-			active={active}
-			closeOnClickOutside
-			menuWidth="shrink"
-			onActiveChange={setActive}
-			trigger={
-				<Button
-					className="align-items-center bg-white d-flex justify-content-between w-100"
-					displayType="secondary"
-					outline
-					small
-				>
-					<div className="text-truncate">
-						{i18n.translate(getKebabCase(selectedAccountRoleName))
-							? i18n.translate(
-									getKebabCase(selectedAccountRoleName)
-							  )
-							: selectedAccountRoleName}
-					</div>
+			return previousItem;
+		}, { partnerMemberRoles: { active: undefined, roles: [] } }))
```

Comment:
I think this line should be expanded into new lines

Reply:
Fixed code formatting.